code/game/objects/items/mop.dm

```diff
@@ -24,16 +26,20 @@
 	create_reagents(mopcap)
 
 
-/obj/item/mop/proc/clean(turf/A)
+/obj/item/mop/proc/clean(turf/A, mob/user)
+	var/cleaned = FALSE
 	if(reagents.has_reagent("water", 1) || reagents.has_reagent("holywater", 1) || reagents.has_reagent("vodka", 1) || reagents.has_reagent("cleaner", 1))
 		SEND_SIGNAL(A, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_MEDIUM)
 		for(var/obj/effect/O in A)
 			if(is_cleanable(O))
+				cleaned = TRUE
 				qdel(O)
+	if(cleaned && user && user.mind && user.mind.assigned_role == "Janitor")
+		SSresearch.station_tech.add_points_all(CLEAN_TILE_REWARD)
+		to_chat(user, "<span class='notice'>Your [src] flashes a message that it has gained useful information from eradicating the mess on the floor. Good work.</span>")
```

Comment:
Too spammy, maybe just like a CoD style +10 popup only the janitor can see

Reply:
I suddenly want a janitor point system

Reply:
@Cyberboss Hmmmm what if I incorporated it into the existing "you clean the floor" message ?